spark/src/main/java/org/apache/iceberg/spark/source/ReaderUtils.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.source;
+
+import org.apache.iceberg.Snapshot;
+import org.apache.iceberg.SnapshotSummary;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.spark.SparkSchemaUtil;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * common reader code between spark2 and spark3 module
+ */
+public class ReaderUtils {
+  private ReaderUtils() {}
+
+  /**
+   * guess table size based on table schema and total records.
+   *
+   * @param table        iceberg table
+   * @param totalRecords total records in the table
+   * @return approxiate size based on table schema
+   */
+  public static long approximateTableSize(StructType tableSchema, long totalRecords) {
```

Comment:
Could you move this method to `SparkSchemaUtil.estimateSize`? I think it makes sense to put it in the schema utility methods, since it relates to a Spark schema. Looks like we don't have a test suite for that, so you could start `TestSparkSchemaUtil` with the tests for this.

Reply:
ack this makes more sense thanks for suggestion